src/System.Management.Automation/security/SecuritySupport.cs

```diff
@@ -1351,9 +1351,8 @@ internal static int Init()
                 string hostname;
                 try
                 {
-                    var processModule = PsUtils.GetMainModule(currentProcess);
-                    hostname = string.Concat("PowerShell_", processModule.FileName, "_",
-                        processModule.FileVersionInfo.ProductVersion);
+                    hostname = string.Concat("PowerShell_", Environment.ProcessPath, "_",
+                        currentProcess.MainModule.FileVersionInfo.ProductVersion);
```

Comment:
In a further PR I will remove the need to use `Process.MainModule.FileVersionInfo` here to avoid loading System.Diagnostics.FileVersionInfo.dll assembly on startup on Windows.